feature(maverick): initial demo screen

kateinkim opened the pull request:
## Please verify the following:

- [ ] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [ ] `README.md` has been updated with your changes, if relevant

## Describe your PR

Per issue #1988, The Components Screen is added to Demo Screen

### What's changed

1. Adds `Tab Navigator` to the demo screen : The icons and their names are not updated yet.
2. Adds `Drawer Layout` to Components Screen : initial screen with styling is done. 
3. Adds animated button to the drawer

### Photos and Videos

|Component Screen|Drawer|Button|
|-|-|-|
|<img width="370" alt="CleanShot 2022-08-11 at 16 35 36@2x" src="https://user-images.githubusercontent.com/53795920/184270971-ed22581b-2af6-40e9-9320-554dc312ef5f.png">|<img width="357" alt="CleanShot 2022-08-11 at 16 35 50@2x" src="https://user-images.githubusercontent.com/53795920/184270988-6f7f548a-9f38-4657-bc9e-02c0477e1bba.png">|![CleanShot 2022-08-11 at 16 52 05](https://user-images.githubusercontent.com/53795920/184271044-b01e5c50-bb82-4fbe-802c-a71ea7d699f2.gif)|

### Issues

1. Since the lists are sorted as sections it might be better to use `SectionList` with `scrollToLocation` than `FlatList` with `scrollToItem` to scroll to the right sections.

2. Due to the `onDrawerOpen` and `onDrawerClose` callbacks in DrawerLayout getting called after the transition the buttons appear slower due to delay.

<img src="https://user-images.githubusercontent.com/53795920/184271057-0035105d-1a47-4ac1-88fc-9c6baf765eac.gif" width="250">

yulolimum commented on boilerplate/app/screens/DemoComponentsScreen/DemoComponentsScreen.tsx:
```diff
@@ -0,0 +1,126 @@
+    >
+      <Screen preset="fixed">
+        <DrawerIconButton open={open} onPress={toggleDrawer} />
+        <View style={$heading}>
+          <Text preset="heading">Components to jump start your project!</Text>
+        </View>
```
Let's move this to FlatList's `ListHeaderComponent` prop so that it scrolls out of the way. Right now, it takes up half the screen on smaller devices.

yulolimum replied:
![yulolimum-capture-2022-08-12--05-03-09](https://user-images.githubusercontent.com/1775841/184350166-d995feb1-1497-440a-89a4-86741de4ac0f.gif)

yulolimum commented on boilerplate/app/screens/DemoComponentsScreen/DemoComponentsScreen.tsx:
```diff
@@ -0,0 +1,126 @@
+          data={Object.values(Demos)}
+          keyExtractor={(item) => item.name}
+          renderItem={(ItemProps) => <DemoItem {...ItemProps} />}
+        />
```
Right now, the FlatList renders much further down below the viewport because of the above View/Text. You'll want to add a flex style to the FlatList.

yulolimum commented on boilerplate/app/screens/DemoComponentsScreen/DemoItem.tsx:
```diff
@@ -0,0 +1,34 @@
+}
+
+const $demoItemContainer: ViewStyle = {
+  marginLeft: 24,
```
you'll want to use `paddingHorizontal` here to contain the content equally.
![yulolimum-capture-2022-08-12--05-05-35@2x](https://user-images.githubusercontent.com/1775841/184350571-048046d1-9b1a-412a-955a-7de3d18a399d.jpg)

yulolimum commented on boilerplate/app/screens/DemoComponentsScreen/DemoComponentsScreen.tsx:
```diff
@@ -0,0 +1,126 @@
+                  {item.useCases.map((u) => (
+                    <View key={u} style={$subMenuItem}>
+                      <Text>{u}</Text>
+                      <Image source={require("../../../assets/images/caratRight.png")} />
```
Minor nitpick - let's change `carat` -> `caret`, although a diamond icon would be kinda cool. :) 

Also, we have an `Icon` component you can use which includes the assets. You can move the icon there for now -  I'll move the assets around later when i work on the component.

yulolimum reviewed (commented):
@kateinkim 

Great job!

A couple suggestions in comments but otherwise i like the direction!

kateinkim commented:
Just added styles to tab navigator and renamed the tabs. You can see `ListHeaderComponent` is working as well.

<img src="https://user-images.githubusercontent.com/53795920/184449360-9d6a6dcb-b2f2-4903-a62d-0af14be7cc70.gif" width="250">

And I don't know if it's just me or if you see this as well but to me, the tab names' weight seems a bit bolder than the ones in the app. The design says their weight is "300" just as the text in `Next Section` description.

|Design|App|
|-|-|
|<img src="https://user-images.githubusercontent.com/53795920/184449896-a2d8e98a-3fc3-4d5c-b352-b177c9bf4621.png">|<img  alt="CleanShot 2022-08-12 at 16 03 49@2x" src="https://user-images.githubusercontent.com/53795920/184450609-39629bf3-6c08-4d54-b97e-1df05845c517.png">|

yulolimum commented:
>And I don't know if it's just me or if you see this as well but to me, the tab names' weight seems a bit bolder in design than the ones in the app. The design says their weight is "300" the same weight as the text in Next Section description.

@kateinkim 

Good question! I think we just need to add some styles to the tab-bar-label. Here's the [commit](https://github.com/infinitered/ignite/pull/2002/commits/a9a5876e4dd92d5bb4c49b3ecfee02df68d862fb) where I did that if you want to take a look.

kateinkim commented on boilerplate/app/navigators/demo-navigator.tsx:
```diff
@@ -69,3 +67,15 @@ export function DemoNavigator() {
+  borderTopColor: colors.transparent,
+}
+
+const $tabBarLabel: TextStyle = {
```
Thanks for listening to my weird concern 😄 👍🏼
